src/DebtRequest.hs

```diff
@@ -123,7 +123,7 @@ mkValidator contractInfo@ContractInfo{..} dat borrowerTn ctx = validate
       collateralAmnt dat <= assetClassValueOf (txOutValue txo) (collateral dat)
 
     containsNewDatum :: TxOut -> Bool
-    containsNewDatum txo = case U.getUpperBound ctx of
+    containsNewDatum txo = case U.getLowerBound ctx of
       Just ub -> findDatumHash' (expectedNewDatum ub) (U.info ctx) == txOutDatumHash txo
```

Comment:
np:
```suggestion
      Just lb -> findDatumHash' (expectedNewDatum lb) (U.info ctx) == txOutDatumHash txo
```

Reply:
Just for clarification, "np" means a nitpick. It can also mean "no problem" and it took me a while to get what @cptmikx means.